scripts/screeps_tencent_batch_rl_runner.py

```
@@ -2334,11 +2356,44 @@ def classify_remote_training_failure(
     if process_failure_class in {"network_unreachable", "host_key_self_healing_failed", "host_key_mismatch"}:
         return process_failure_class
+    simulator_setup_text = "\n".join(
+        tail
+        for filename, entry in files.items()
+        for tail in [entry.get("tail")]
+        if isinstance(tail, str)
+        and (
+            filename.endswith("setup_failure.json")
+            or filename.endswith("run_summary.json")
+            or filename.endswith("run_failure.json")
+            or REMOTE_SIMULATOR_SETUP_CONTEXT_RE.search(tail)
```

Ruff (0.15.14)

[warning] Call `endswith` once with a `tuple`

Merge into a single `endswith` call

(PIE810)